### PR TITLE
OSDOCS 10594: Updated the user defined labels and tags for GCP section with a note, and other changes

### PR DESCRIPTION
**Pull request by subhtk**
Version(s): 4.16

Issue: [OSDOCS-9804](https://issues.redhat.com/browse/OSDOCS-9804)

Link to docs preview: [Preview](https://73001--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations#installing-gcp-user-defined-labels-and-tags_installing-gcp-customizations)

SME review:
- [x] SME has approved this change.

Additional information:

### CODE REVIEW

**Comment by bharath-b-rh** [comment]
Input was not clear and caused this, but points at lines 63, 64, 65 are required. Otherwise lgtm.

**Comment by subhtk** [comment]
Let me restore the content.

**Review by adellape — COMMENTED** [review]
The PR title says "Removed tech preview" but I see the following still in the file:

```
:FeatureName: Support for user-defined labels and tags for GCP
include::snippets/technology-preview.adoc[]
```

Should that be removed? And will there be a separate PR to similarly update the 4.16 release notes?

Also, I see the linked Jira tickets reference OCP 4.17 (`Fix version/s: openshift-4.17`) but this PR is targeting the `enterprise-4.16` branch. Two thoughts:

1. Are those Jiras actually for 4.16 instead? If so, it seems like this PR should target `main` and then get cherrypicked to `enterprise-4.16`.
2. If the Jiras are truly not relevant until 4.17, then maybe this PR should just target `main` and you have a follow-up task later to cherry-pick it to the future `enterprise-4.17` branch (that isn't created yet and we also don't have a GH label for yet)? Or maybe just leave it unmerged for now and pick it back up after the 4.17 branch has been created.

File: modules/installing-gcp-user-defined-labels-and-tags.adoc [hunk 1]
```diff
@@ -17,6 +17,11 @@ You can define labels and tags for each GCP resource only during {product-title}
 User-defined labels and tags are not supported for {product-title} clusters upgraded to {product-title} {product-version}.
 ====
 
+[NOTE]
+====
+Tags once added cannot be updated. Also, a new tag-supported resource creation would fail if configured tags are deleted (key or value).
```
[the remark this answers is not in the document]
Comment (subhtk) [reply modules/installing-gcp-user-defined-labels-and-tags.adoc:22]: Ack.

File: modules/installing-gcp-user-defined-labels-and-tags.adoc [hunk 2]
```diff
@@ -44,26 +49,26 @@ User-defined labels and {product-title} labels are available on the following GC
 
 .User-defined tags
 
-User-defined tags are attached to resources created by the {product-title} Image Registry Operator and not on the resources created by any other Operators or the Kubernetes in-tree components.
+User-defined tags are attached to resources created by the {product-title} Installer, Image Registry Operator, Machine API Operator and not on the resources created by any other Operators or the Kubernetes in-tree components.
```
[the remark this answers is not in the document]
Comment (subhtk) [reply modules/installing-gcp-user-defined-labels-and-tags.adoc:52]: ACK.

File: modules/installing-gcp-user-defined-labels-and-tags.adoc [hunk 3]
```diff
@@ -44,26 +49,26 @@ User-defined labels and {product-title} labels are available on the following GC
...
 
 User-defined tags are available on the following GCP resources:
+
 * Storage bucket
+* Compute Instances
+* Compute Disks
```
[the remark this answers is not in the document]
Comment (subhtk) [reply modules/installing-gcp-user-defined-labels-and-tags.adoc:58]: ack

**Comment by subhtk** [comment]
Regarding the title, I did change the commit message but forgot to change the title, let me change the title too. And regarding the fix versions, when the jira was created, the feature was indeed planned for GA in 4.16 but due to some issues, it was planned for 4.17. So this PR addresses the minor updates for 4.16.

**Review by adellape — APPROVED** [review]
Thank you, explanations sound good to me!

File: modules/installing-gcp-user-defined-labels-and-tags.adoc [hunk 4]
```diff
@@ -17,6 +17,11 @@ You can define labels and tags for each GCP resource only during {product-title}
 User-defined labels and tags are not supported for {product-title} clusters upgraded to {product-title} {product-version}.
 ====
 
+[NOTE]
+====
+After adding tags, they cannot be updated. Also, a new tag-supported resource creation would fail if configured tags are deleted (key or value).
```
Comment (snarayan-redhat) [inline modules/installing-gcp-user-defined-labels-and-tags.adoc:22]:
```suggestion
You cannot update the tags that are already added. Also, a new tag-supported resource creation fails if the configured tag keys or tag values are deleted.
```

File: modules/installing-gcp-user-defined-labels-and-tags.adoc [hunk 5]
```diff
@@ -44,26 +49,26 @@ User-defined labels and {product-title} labels are available on the following GC
 
 .User-defined tags
 
-User-defined tags are attached to resources created by the {product-title} Image Registry Operator and not on the resources created by any other Operators or the Kubernetes in-tree components.
+User-defined tags are attached to resources created by the {product-title} Installer, Image Registry Operator, and Machine API Operator. User-defined tags are not attached to resources created by any other Operators or the Kubernetes in-tree components.
```
Comment (snarayan-redhat) [inline modules/installing-gcp-user-defined-labels-and-tags.adoc:52]:
```suggestion
User-defined tags are attached to resources created by the {product-title} installation program, Image Registry Operator, and Machine API Operator. User-defined tags are not attached to the resources created by any other Operators or the Kubernetes in-tree components.
```

**Review by snarayan-redhat — COMMENTED** [review]
Couple more comments.

File: modules/installing-gcp-user-defined-labels-and-tags.adoc (hunk 4 above)
Comment (subhtk) [reply modules/installing-gcp-user-defined-labels-and-tags.adoc:22]: Thanks, Ack...

File: modules/installing-gcp-user-defined-labels-and-tags.adoc (hunk 5 above)
Comment (subhtk) [reply modules/installing-gcp-user-defined-labels-and-tags.adoc:52]: Ack.

**Comment by subhtk** [comment]
> The PR title says "Removed tech preview" but I see the following still in the file:
> 
> ```
> :FeatureName: Support for user-defined labels and tags for GCP
> include::snippets/technology-preview.adoc[]
> ```
> 
> Should that be removed? And will there be a separate PR to similarly update the 4.16 release notes?
> 
> Also, I see the linked Jira tickets reference OCP 4.17 (`Fix version/s: openshift-4.17`) but this PR is targeting the `enterprise-4.16` branch. Two thoughts:
> 
>     1. Are those Jiras actually for 4.16 instead? If so, it seems like this PR should target `main` and then get cherrypicked to `enterprise-4.16`.
> 
>     2. If the Jiras are truly not relevant until 4.17, then maybe this PR should just target `main` and you have a follow-up task later to cherry-pick it to the future `enterprise-4.17` branch (that isn't created yet and we also don't have a GH label for yet)? Or maybe just leave it unmerged for now and pick it back up after the 4.17 branch has been created.

Since the jira issue is about GA and the PR was initially created to track the change, but since the affected version is 4.17, and the changes are irrelevant to tech preview. I have created another jira ticket https://issues.redhat.com/browse/OSDOCS-10594 to track the changes and here is the new PR- https://github.com/openshift/openshift-docs/pull/76127. Sorry for the confusion.

**Comment by subhtk** [comment]
Due to the version change, I have created a new PR-https://github.com/openshift/openshift-docs/pull/76141, and will be closing this one.

**Pull request closed without merging**